File: ts/src/test/base/test.omit.ts

// AUTO_TRANSPILE_ENABLED

import ccxt from '../../../ccxt.js';
import testSharedMethods from '../Exchange/base/test.sharedMethods.js';

function testOmit () {

    const exchange = new ccxt.Exchange ({
        'id': 'sampleexchange',
    });

    testSharedMethods.assertDeepEqual (exchange, undefined, 'testOmit',  exchange.omit ({ }, 'foo'), {});
    testSharedMethods.assertDeepEqual (exchange, undefined, 'testOmit',  exchange.omit ({ 'foo': 2 }, 'foo'), { });
    testSharedMethods.assertDeepEqual (exchange, undefined, 'testOmit',  exchange.omit ({ 'foo': 2, 'bar': 3 }, 'foo'), { 'bar': 3 });
    testSharedMethods.assertDeepEqual (exchange, undefined, 'testOmit',  exchange.omit ({ 'foo': 2, 'bar': 3 }, [ 'foo' ]), { 'bar': 3 });
    // todo: below will be added later
    // testSharedMethods.assertDeepEqual (exchange, undefined, 'testOmit',  exchange.omit ({ 'foo': 2, 'bar': 3 }), { 'foo': 2, 'bar': 3 }); // todo: bugs in php, ArgumentCountError: Too few arguments to function ccxt\Exchange::omit(), 1 passed in 2 expected
    // testSharedMethods.assertDeepEqual (exchange, undefined, 'testOmit',  exchange.omit ({ 'foo': 2, 'bar': 3 }, 'foo', 'bar'), {}); // todo: bugs in php
    // testSharedMethods.assertDeepEqual (exchange, undefined, 'testOmit',  exchange.omit ({ 'foo': 2, 'bar': 3 }, [ 'foo' ], 'bar'), {}); // todo: bugs in php
    // testSharedMethods.assertDeepEqual (exchange, undefined, 'testOmit',  exchange.omit ({ '5': 2, 'bar': 3 }, [ 5 ]), { 'bar': 3 }); // todo: bugs in py
    // testSharedMethods.assertDeepEqual (exchange, undefined, 'testOmit',  exchange.omit ({ '5': 2, 'bar': 3 }, 5), { 'bar': 3 }); // todo: bugs in py
}

export default testOmit;

File: julia/Ccxt/test/base/test.omit.jl
using Test
using Ccxt
function testOmit()

    exchange = get(ccxt, Symbol("Exchange"), nothing)(Dict{Symbol, Any}(
        Symbol("id") => "sampleexchange"
    ));
    assertDeepEqual(testSharedMethods, exchange, nothing, "testOmit", omit(exchange, Dict{Symbol, Any}(), "foo"), Dict{Symbol, Any}());
    assertDeepEqual(testSharedMethods, exchange, nothing, "testOmit", omit(exchange, Dict{Symbol, Any}(
    Symbol("foo") => 2
), "foo"), Dict{Symbol, Any}());
    assertDeepEqual(testSharedMethods, exchange, nothing, "testOmit", omit(exchange, Dict{Symbol, Any}(
    Symbol("foo") => 2,
    Symbol("bar") => 3
), "foo"), Dict{Symbol, Any}(
    Symbol("bar") => 3
));
    assertDeepEqual(testSharedMethods, exchange, nothing, "testOmit", omit(exchange, Dict{Symbol, Any}(
    Symbol("foo") => 2,
    Symbol("bar") => 3
), ["foo"]), Dict{Symbol, Any}(
    Symbol("bar") => 3
));
end
